ethers/wallet.nim

```diff
@@ -0,0 +1,140 @@
+import eth/keys
+import eth/rlp
+import eth/common
+import eth/common/transaction as ct
+import stew/byteutils
+import ./providers/jsonrpc
+import ./transaction
+import ./signer
+
+export keys
+
+let rng = newRng()
+
+type SignableTransaction = common.Transaction
+
+type WalletError* = object of EthersError
+type Wallet* = ref object of Signer
+  privateKey*: PrivateKey
+  publicKey*: PublicKey
+  address*: Address
+  provider*: ?JsonRpcProvider
```

Comment:
Ideally, the Wallet wouldn't know about specific providers such as the `JsonRpcProvider`, only about the base `Provider` type. This would require defining `sendRawTransaction` on the base `Provider` type.